In [59]:
import numpy as np
import scipy.stats as st
from IPython.display import Image

4.10 Twitter users and news, Part II. A poll conducted in 2013 found that 52% of U.S.
adult Twitter users get at least some news on Twitter, and the standard error for this estimate
was 2.4%. Identify each of the following statements as true or false. Provide an explanation to
justify each of your answers.

(a) The data provide statistically significant evidence that more than half of U.S. adult Twitter
users get some news through Twitter. Use a significance level of $\alpha$ = 0.01.

(b) Since the standard error is 2.4%, we can conclude that 97.6% of all U.S. adult Twitter users
were included in the study.

(c) If we want to reduce the standard error of the estimate, we should collect less data.

(d) If we construct a 90% confidence interval for the percentage of U.S. adults Twitter users who
get some news through Twitter, this confidence interval will be wider than a corresponding
99% confidence interval.

In [100]:
mean = 0.52
s_e = 0.024

In [101]:
# a.

# H_0 : mean <= 0.5
# H_a : mean > 0.5

# We have an upper tail test

alpha = 0.01

z_critical = st.norm.ppf(q = (1 - alpha / 2)) 

ci_99 = s_e * z_critical

print("Our value with confidence 99 percent interval is %.4f +/- %.4f" % (mean, ci_99))

[mean - ci_99, mean + ci_99]

Our value with confidence 99 percent interval is 0.5200 +/- 0.0618


[0.4581800967148264, 0.5818199032851736]

In [102]:
1 - st.norm.cdf(0.5, loc=mean, scale=s_e)

0.7976716190363572

The answer to question a. is that does not provide statistically significant evidence that more than half of U.S. adult Twitter users get some news through Twitter as the 50% falls within the C.I. and P(X > 0.5) = 1 - P(X <= 0.5) =  0.7976 > 0.01

b.

False. This is not an interpretation of standard error, we don't know how many people were included in the study. Therefore, we cannot make any solid conclusion.

In [104]:
alpha_975 = 1 - 0.975

z_critical_975 = st.norm.ppf(q = (1 - alpha_975 / 2)) 


ci_975 = s_e * z_critical_975

print("Our value with confidence 97.5 percent interval is %.4f +/- %.4f" % (mean, ci_975))
[mean - ci_975, mean + ci_975]

Our value with confidence 97.5 percent interval is 0.5200 +/- 0.0538


[0.4662063345374813, 0.5737936654625188]

In [105]:
n_up = 0.5737936654625188 * (1 - 0.5737936654625188) / 0.024**2
print(n_up)

n_down = 0.4662063345374813 * (1 - 0.4662063345374813) / 0.024**2
print(n_down)

424.5737759333435
432.04511835869334


So, yeah, this does not display 97.5% of adult population at all.

c.

False. If we want to reduce the standard error of the estimate, we should increase the amound of data. Let's take a look at the example below.

We know that the \begin{equation*} s.e. =  \sqrt{ \frac{\widehat{p} \times (1 - \widehat{p})}{n} } \end{equation*}

From the above, we can derive the sample size n

In [106]:
n = 0.52 * (1 - 0.52) / 0.024**2
n

433.3333333333333

In [107]:
# Then

s_e_500 = np.sqrt(0.52 * (1 - 0.52) / 500)

print(s_e_500)

s_e_1000 = np.sqrt(0.52 * (1 - 0.52) / 1000)

print(s_e_1000)

s_e_10000 = np.sqrt(0.52 * (1 - 0.52) / 10000)

print(s_e_10000)

0.022342784070030305
0.0157987341265052
0.004995998398718718


In [108]:
# d.

alpha_90 = 1- 0.9
alpha_99 = 1- 0.99

z_critical_90 = st.norm.ppf(q = (1 - alpha_90 / 2)) 
z_critical_99 = st.norm.ppf(q = (1 - alpha_99 / 2)) 


ci_90 = s_e * z_critical_90
ci_99 = s_e * z_critical_99

print("Our value with confidence 90 percent interval is %.4f +/- %.4f" % (mean, ci_90))
print("Our value with confidence 99 percent interval is %.4f +/- %.4f" % (mean, ci_99))

Our value with confidence 90 percent interval is 0.5200 +/- 0.0395
Our value with confidence 99 percent interval is 0.5200 +/- 0.0618


The answer to part d. is False.

4.20 

Age at first marriage, Part II. Exercise 4.16 presents the results of a 2006 - 2010 survey
showing that the average age of women at first marriage is 23.44. Suppose a social scientist believes
that this value has increased in 2012, but she would also be interested if she found a decrease.
Below is how she set up her hypotheses. Indicate any errors you see.

$H_0$ : $\tilde{x}$ = 23.44 years old

$H_A$: $\tilde{x}$  > 23.44 years old

The value of the mean should be μ and alternate should be a ≠

4.30

Testing for food safety. A food safety inspector is called upon to investigate a restaurant with a few customer reports of poor sanitation practices. The food safety inspector uses a hypothesis testing framework to evaluate whether regulations are not being met. If he decides the restaurant is in gross violation, its license to serve food will be revoked.

(a) Write the hypotheses in words.

(b) What is a Type 1 Error in this context?

(c) What is a Type 2 Error in this context?

(d) Which error is more problematic for the restaurant owner? Why?

(e) Which error is more problematic for the diners? Why?

(f) As a diner, would you prefer that the food safety inspector requires strong evidence or very
strong evidence of health concerns before revoking a restaurant’s license? Explain your reasoning.

a. 

$H_0$ : Food sanitation is in good condition.

$H_a$ : Food sanitation is not in good condition.

b. 

Type 1 Error (False Positive) is rejecting the null hypothesis when it is actualy True. This means that we reject the null hypothesis in favour of restaurant not meeting food sanitation requirements when it is actually meeting them.

c.

Type 2 Error (False Negative) is failing to reject null hypothesis when it is actually False. This means that we don't reject null hypothesis in favour of restaurant meeting food sanitation requirements when it actually does not meet the food sanitation requirements.

d.

Type 1 Error would be more problematic to the restaurant owner as his/her restaurant would be shut down by inspection as the inspection would conclude that the restaurant does not meet sanitation requirements when it actually does. Therefore, restaurant would be erroneously closed down for violating sanitation regulations.

e.

Type 2 Error would be more problematic to the customers as sanitation inspection should have closed down restaurant when it did not due to the restaurant violating the sanitation requirements. Therefore, customers would be eating unsafe food.

f.

As a diner, I would prefer food inspector have strong evidence ($\alpha$ not too big and not too small, around 0.05). If $\alpha$ is too small (0.01), meaning there is very strong evidence, there is a higher chance of Type 2 error and as a diner we would not want that. If $\alpha$ is larger but not too big (0.10), meaning there is some evidence, there is a lower chance of Type 2 error and as a diner I would be more content with that; here we want to be cautious about failing to reject $H_0$ when the null is actually false..

5.4

An independent random sample is selected from an approximately normal population with an unknown standard deviation. Find the p-value for the given

set of hypotheses and T test statistic. Also determine if the null hypothesis would be rejected at
$\alpha$ = 0.01.

(a) $H_a$ : μ > 0.5, n = 26, T = 2.485; upper tail 

(b) $H_a$ : μ < 3, n = 18, T = 0.5; lower tail

In [109]:
st.t.ppf(0.99, 25)

2.4851071754106413

In [110]:
# a

print(st.t.ppf(0.99, 25))

print(1 - st.t.cdf(2.485, 25))

2.4851071754106413
0.010002400548908885


Do not reject the null hypothesis since p-value = 0.010002400548908885 > $\alpha$ = 0.01

In [111]:
print(st.t.ppf(0.01, 25))

print(st.t.cdf(0.5, 17))

-2.4851071754106417
0.6882573967171443


Do not reject the null hypothesis since p-value = 0.6882573967171443 >  α  = 0.01

5.8

Fuel efficiency of Prius. Fueleconomy.gov, the offcial US government source for fuel economy information, allows users to share gas mileage information on their vehicles. The histogram below shows the distribution of gas mileage in miles per gallon (MPG) from 14 users who drive a 2012 Toyota Prius. The sample mean is 53.3 MPG and the standard deviation is 5.2 MPG. Note that these data are user estimates and since the source data cannot be verified, the accuracy of these estimates are not guaranteed.

<img src="homework58.png">

(a) We would like to use these data to evaluate the average gas mileage of all 2012 Prius drivers.
Do you think this is reasonable? Why or why not?

(b) The EPA claims that a 2012 Prius gets 50 MPG (city and highway mileage combined). Do these data provide strong evidence against this estimate for drivers who participate on fueleconomy.gov? Note any assumptions you must make as you proceed with the test.

(c) Calculate a 95% confidence interval for the average gas mileage of a 2012 Prius by drivers who
participate on fueleconomy.gov.

a. 

In order to evaluate the average gas mileage of all 2012 Prius drivers we need to ensure that our sample data is independently and identically distributed Random Normal. From the graph we can see that the sample data is approximately Normally distributed with little to no skew. We know that the sample is less that 10% of the population. What we do not know whether our sample is randomly chosen or not since users volunteerily submitted their estimates which can lead to biased results. Additionally, to ensure that our data is normally distributed, we need at least 30 data entries. Therefore, we need to be extremely sceptical when trying to deal with this data.

b.

Let $\mu$ be the average MPG a 2012 Prius gets both (city and highway combined).

$ H_0: \mu = 50 MPG $

$H_a: \mu \neq  50 MPG $

Then, we need to have the following assumptions:

- Data seems to be Normally distributed with little to no skew

- Size of the sample is less than 30, therefore will use t-distribution.

- Data seems to be symmetrically distributed.

- Gas mileage entries are independent of each other

- $\alpha$ = 0.05

In [116]:
sample_mean = 53.3
sample_standard_deviation = 5.2
sample_size = 14
degrees_of_freedom = sample_size - 1
sample_s_e = sample_standard_deviation / np.sqrt(sample_size)

T = st.t.ppf(1-0.05/2, degrees_of_freedom)

print(T)

t_stat = (sample_mean - 50) / sample_s_e

print(t_stat)

calculated_proba = 2*(1 - st.t.cdf(t_stat, 13))

print(calculated_proba)

2.1603686564610127
2.3745133416065376
0.03365111175830604


Reject Null Hypothesis since 2 $\times$ p-value = 2 $\times$ 0.01682555587915302 = 0.03365111175830604 < $\alpha$ = 0.05

Also, t-score = 2.3745133416065376 > $t_{0.05/2, 13}$ = 2.1603686564610127 and t-score = 2.3745133416065376 ! < $-t_{0.05/2, 13}$ = -2.1603686564610127

In [117]:
# c.

small_ci = st.t.ppf(1-0.05/2, degrees_of_freedom) * sample_s_e

print("Our value with confidence interval is %.4f +/- %.4f" % (sample_mean, small_ci))
[sample_mean - small_ci, sample_mean + small_ci]

Our value with confidence interval is 53.3000 +/- 3.0024


[50.297609387404876, 56.30239061259512]

Additionally $\mu$ = 50 does not fall into the C.I.

5.20

High School and Beyond, Part I. The National Center of Education Statistics conducted
a survey of high school seniors, collecting test data on reading, writing, and several other subjects.
Here we examine a simple random sample of 200 students from this survey. Side-by-side box plots
of reading and writing scores as well as a histogram of the differences in scores are shown below.

<img src="homework520.png">

(a) Is there a clear difference in the average reading and writing scores?

(b) Are the reading and writing scores of each student independent of each other?

(c) Create hypotheses appropriate for the following research question: is there an evident difference
in the average scores of students in the reading and writing exam?

(d) Check the conditions required to complete this test.

(e) The average observed difference in scores is $\tilde{x}_{read-write}$ =
-0.545, and the standard deviation
of the differences is 8.887 points. Do these data provide convincing evidence of a difference
between the average scores on the two exams?

(f) What type of error might we have made? Explain what the error means in the context of the
application.

(g) Based on the results of this hypothesis test, would you expect a confidence interval for the
average difference between the reading and writing scores to include 0? Explain your reasoning.

a.

By looking at the above graphs, we believe that there is not a difference in the average reading and writing scores. The difference distributions seems to be fairly normal with mean around 0.

b.

Reading scores and writing scores of one student are not independent of each other although the scores between students would be independent. One's ability to read would affect one's ability to write and perhaps vice versa. Therefore, we have paired data.

c.

$H_0 : \mu_{reading - writing} = 0$

$H_a : \mu_{reading - writing} \ne 0$

d.

- Independence
    - We have independence since our sample is random as stated in the problem definition. Additionally, sample size seems to be less that 10% of the disribution.
- Normality
    - By looking at the box plots, we can see that both writing scores and reading scores are relatively normally distributed, although writing seems to be skewed a bit to the left due to a longer lower tail.
- Randomness
    - As per the problem definition, we have a simple random sample. 

e.

$\tilde{x}_{reading - writing}$ = -0.545

$s_{reading - writing}$ = 8.887

In [120]:
x_bar = -0.545
hypothesis_mean = 0
sample_size = 200
sample_standard_deviation = 8.887
standard_err = sample_standard_deviation / np.sqrt(sample_size)
alpha = 0.05

T = st.t.ppf(1 - alpha / 2, sample_size -1 )

print("T:", T)

t_score = (x_bar - hypothesis_mean) / standard_err

print("t_score: ", t_score)

p_value = st.t.cdf(t_score, sample_size - 1)

print("p_value: ", p_value)

T: 1.971956544249395
t_score:  -0.8672739861520613
p_value:  0.19341823709967354


Do not reject the null hypothesis since:

- t-score = -0.8672739861520613 > = T = -1.971956544249395

- 2 $\times$ p_value = 2 $\times$ 0.19341823709967354 = 0.3868364741993471 > $\alpha$ = 0.05

Therefore, there is no difference between the average scores on two exames

f.

Since we did not reject the null hypothesis when it could have been erroneous, we might have made a Type 2 error. Meaning, we failed to reject the null hypothesis in favour of there being no difference between the scores when there could actually be difference between the scores.

g.

Since we did not reject the null hypothesis, we do expect the CI to contain 0.

In [115]:
alpha_95 = 1- 0.95

t_critical_95 = st.t.ppf(1 - alpha_95 / 2, sample_size -1 )

ci_95 = standard_err * t_critical_95

print("Our value with confidence 90 percent interval is %.4f +/- %.4f" % (x_bar, ci_95))
[x_bar - ci_95, x_bar + ci_95]

Our value with confidence 90 percent interval is 0.5450 +/- 1.2392


[-0.6941889227350669, 1.784188922735067]